tap_snowflake/client.py

```diff
@@ -91,13 +113,22 @@ def discover_catalog_entries(self) -> list[dict]:
         tables = [t.lower() for t in self.config.get("tables", [])]
         engine = self.create_sqlalchemy_engine()
         inspected = sqlalchemy.inspect(engine)
-        schema_names = [
-            schema_name
-            for schema_name in self.get_schema_names(engine, inspected)
-            if schema_name.lower() != "information_schema"
-        ]
+
+        if self.config.get("schema"):
```

Comment:
Hi, is there an update on this? We would really like to merge this in asap.

Reply:
@edgarrmondragon ping on the previous message in case you have thoughts but we mentioned it in office hours and it looks like this is implementation specific and not related to the SDK as I originally thought.

@nidhi-akkio sorry about the delay. A few more comments but I think we can get this merged soon.

This is a of breaking change for some users who assigned a schema to use for connecting but are expecting the status quo where its not using it for filtering. If they upgrade they'll silently stop syncing tables outside this schema. We can either:

1. accept the breaking change and document it well in release notes and update the tap setting description which will also get pulled into the hub docs.
2. we can add this as a new config like `schema_filter`. Avoiding breaking changes but likely making the difference between the 2 options confusing for users, although `schema` in its current state is already confusing to me 😅 .

@nidhi-akkio @edgarrmondragon what do you think?

Reply:
I think, like you say @pnadolny13, that the current usage and description of the `schema` setting, i.e. `The initial schema for the Snowflake session.` is rather confusing.

My preference would be to not introduce a breaking change, add a new setting `schema_filter` and think about what to do with the existing `schema` setting.

cc @nidhi-akkio